In [1]:
import torch
from torchvision import models, transforms
from PIL import Image

# Load the pretrained ResNet50 model using 'weights' instead of 'pretrained'
model = models.resnet50(weights=models.ResNet50_Weights.IMAGENET1K_V1)  # Updated line
model.eval()  # Set the model to evaluation mode

# Define preprocessing for the input image
preprocess = transforms.Compose([
    transforms.Resize(256),          # Resize the image to 256x256
    transforms.CenterCrop(224),      # Crop the center 224x224
    transforms.ToTensor(),           # Convert to Tensor
    transforms.Normalize(            # Normalize with ImageNet stats
        mean=[0.485, 0.456, 0.406],
        std=[0.229, 0.224, 0.225]
    )
])

# Load and preprocess the image
image = Image.open("/content/Object.png")  # Replace with your image path
input_tensor = preprocess(image).unsqueeze(0)  # Add batch dimension

# Perform inference
with torch.no_grad():
    outputs = model(input_tensor)

# Get the predicted class
_, predicted_class = outputs.max(1)  # Get the index of the max logit
print(f"Predicted class index: {predicted_class.item()}")


Predicted class index: 870


Predicted class index 870 refers to the most prominent object in the image, based on the model's classification.
